tests/test_converter.py

```diff
@@ -50,25 +51,39 @@ def test_convert_sample(self):
         # given:
         biomaterial = self.hca_data.get('biomaterial')
         with open(config.JSON_DIR + 'dsp/sample.json', encoding=config.ENCODING) as data_file:
-            expected_json = json.loads(data_file.read())
+            no_release_date = json.loads(data_file.read())
 
         # and:
         test_alias = 'hca' + str(randint(0, 1000))
         biomaterial['uuid']['uuid'] = test_alias
-        expected_json['alias'] = test_alias
-        expected_json['attributes']['HCA Biomaterial UUID'] = [{'value': test_alias}]
+        no_release_date['alias'] = test_alias
+        no_release_date['attributes']['HCA Biomaterial UUID'] = [{'value': test_alias}]
+
+        # and:
+        with_release_date = deepcopy(no_release_date)
+        with_release_date['releaseDate'] = '2018-10-11'
 
         # and:
         converter = SampleConverter(ontology_api=self.ontology_api)
         converter.ingest_api = self.ingest_api
 
         # when:
-        actual_json = converter.convert({
-            'biomaterial': biomaterial
+        converted_with_release_date = converter.convert({
+            'biomaterial': biomaterial,
+            'project': {
+                'releaseDate': '2018-10-11T14:33:22.111Z'
+            }
+        })
+
+        # and:
+        converted_no_release_date = converter.convert({
+            'biomaterial': biomaterial,
+            'project': {}
```

Comment:
I think 
```
'project' : {
   'releaseDate': None
}
```
is the test scenario for when there's an empty releaseDate

Reply:
This is effectively the same. I think it's possible for `releaseDate` filed not be specified at all, hence the `and` check in the conversion code.

Reply:
I guess there are two no-release-date scenarios: 1. when no `releaseDate` key is present, and 2. when `releaseDate`'s value is null. A test scenario for 2. would've caught the bug during unit tests

Reply:
I've updated the test case to use a blank release date instead of `None`. The truthiness are pretty much the same but I guess a blank `releaseDate` field is more likely to come up as a result of JSON parsing.

Reply:
This doesn't make sense to me. The ingest-api specifies releaseDate to be a date-time/Instant, not a string. Since it's not a required field in the ingest-api it's value can be null. I wouldn't expect it to be an empty string, unless the ingest-api's JSON serialization encountered a very strange issue

Reply:
I was in the impression that since dates are sent over as strings in JSON, null dates would appear as empty strings. It looks like Jackson sets them to `null` by default.